res/controllers/Traktor-Kontrol-S4-MK2-hid-scripts.js

```diff
@@ -0,0 +1,1275 @@
+/****************************************************************/
+/*      Traktor Kontrol S4 MK2 HID controller script v1.00      */
+/*      Copyright (C) 2015, the Mixxx Team                      */
+/*      but feel free to tweak this to your heart's content!    */
+/*      For Mixxx version 2.0                                   */
+/****************************************************************/
+
+
+// TODO:
+// * Loop size readout / selection
+// * Find a use for snap / master / other unused buttons?
+
+TraktorS4MK2 = new function() {
```

Comment:
> The other way to do this is the single line: function TraktorS4MK2() {}

@Pegasus-RPG this has slightly different semantics than ```new function() {}```.

See: https://jsfiddle.net/1k13o628/1/

One is a "constructor" to a new object, the other declares a function. IMO what @ywwg switched to in the latest commits (instead of having a "construct" method actually doing that work in the object constructor) is much more idiomatic Javascript.